python/python/bystro/ancestry/tests/test_train.py

```diff
@@ -286,3 +288,72 @@ def test_1kgp_vcf_to_dosage():
     for column in processed_vcf.columns:
         for value in processed_vcf[column]:
             assert value in valid_values
+
+
+def test_restrict_loadings_variants_to_vcf():
+    """Tests restriction of gnomad loadings variants to vcf variants."""
+    num_samples = 10
+    num_vars = 10
+    simulated_vcf = generate_simulated_vcf(num_samples, num_vars)
+    sim_vcf_df = convert_sim_vcf_to_df(simulated_vcf)
+    loaded_vcf = convert_1kgp_vcf_to_dosage(sim_vcf_df)
+    processed_sim_vcf = process_vcf_for_pc_transformation(loaded_vcf)
+    # Using same indices, generate sim loadings
+    num_pcs = 30
+    sim_pcs = np.random.random((num_samples, num_pcs))
```

Comment:
FYI technically sim_pcs is a random semi-orthogonal matrix.  You can simulate this by generating random data, PCAing it, and taking the `pca.components_` matrix.  I don't think it really matters here for what you're doing but if you extend this test later, you should be aware that a PC matrix has special structure that this matrix won't have.

Reply:
I think this is a little further than I'd like to go for this particular test - wasn't planning on making a super realistic simulated gnomad pc loadings since it's limited to just this one part of one module, as compared to a vcf sim that's closer to the real data to test various other analysis modules. Will re-evaluate if it ends up extending further.